docs/src/sections/solution/webapp.adoc

```diff
@@ -0,0 +1,480 @@
+----
+include::../../../../webapp/src/webService/requests.ts[lines=42..51]
+...
+----
+
+Klasa _Requests_ posiada pole _api_, przechowujące funkcje, które wysyłają zapytania do API serwera.
+Każda funkcja jest odpowiedzialna za komunikację z jedną ścieżką serwera.
+
+.Pole _api_ metody _Requests_
+[source, typescript]
+----
+include::../../../../webapp/src/webService/requests.ts[lines=53..68]
+...
+----
+
+Dla ujednolicenia funkcji w polu _api_, każda z nich zwraca ten sam interfejs - _Response_.
+
+.Typ _Response_ pliku _requests.ts_
+[source, typescript]
+----
+include::../../../../webapp/src/webService/requests.ts[lines=31..34]
+----
+
+Każda funkcja w polu _api_ wykorzystuje metodę _request_, która dodaje do zapytania nagłówek z tokenem i wykonuje je przy użyciu biblioteki _axios_.
+
+.Metoda _request_ klasy _Requests_
+[source, typescript]
+----
+include::../../../../webapp/src/webService/requests.ts[lines=195..216]
+----
+
+Ponieważ serwer zwraca różne kody HTTP, a biblioteka _axios_ wyrzuca błąd jeśli kod odpowiedzi nie jest kodem powodzenia, metoda _request_ wyłapuje błąd zwracany przez axios i zwraca obiekt typu _Response_ z pustym polem danych, zamiast wyrzucać błąd. Oznacza to, że nie trzeba wyłapywać błędu przy każdym wywołaniu funkcji pola _api_. Żeby zobaczyć, czy zapytanie zakończyło się powodzeniem, wystarczy sprawdzić pole _status_ zwracanego obiektu.
+
+.Przykład użycia pliku _requests.ts_
+[source, typescript]
+----
+import { requests } from 'webService/requests'
+
+const { data, status } = requests.api.auth.login('admin@admin.pl', 'password1')
+----
+
+==== Portfel
+
+Katalog _src/components/wallet_ zawiera komponenty potrzebne do stworzenia widoku wyświetlającego dane o portfelu.
+
+===== _WalletCard.tsx_
+
+Widok portfelu składa się z kafelek, dlatego powstał komponent _WalletCard.tsx_, z którego korzystają wszystkie komponenty reprezentujące kafelek, aby ujednolicić wygląd i uniknąć powtórzeń.
+
+.WalletCard.tsx
+[source, typescriptjsx]
+----
+include::../../../../webapp/src/components/wallet/WalletCard.tsx[lines=9..16]
+----
+
+.Kafelki na widoku portfelu
+image::../../images/system_presentation/webapp/wallet_cropped.png[]
+
+===== _ProgressCard.tsx_
+
+_ProgressCard.tsx_, podobnie jak _WalletCard.tsx_, jest wykorzystywany w wielu miejscach.
+Przy wyświetlaniu stanu środków portfelu i stanu środków w kanałach _Lightning Network_
+Przyjmuje informacje o tym jaki tekst ma być wyświetlony na kafelce i wartości liczbowe jakie mają być wyświetlone na pasku wczytywania.
+
+.ProgressCard.tsx
+[source, typescriptjsx]
+----
+include::../../../../webapp/src/components/wallet/ProgressCard.tsx[lines=25..41]
+----
+
+Dane liczbowe są przekazywane do komponentu _BalanceProgressBar.tsx_, który jest odpowiedzialny za wyświetlanie paska.
+
+.BalanceProgressBar.tsx
+[source, typescriptjsx]
+----
+include::../../../../webapp/src/components/wallet/BalanceProgressBar.tsx[lines=11..34]
+----
+
+===== _TotalIncomeChart.tsx_
```

Comment:
Asciidoc nie wspiera H5, max możesz dać H4. W tych sekcjach po prostu zrób płynne przejście pomiędzy akapitami, na przykład zdaniem "Przechodząc do komponentu TotalIncomeChart, ...."

Reply:
jak to wygenerujesz jako pdf to wygląda jak gówno